django/http/request.py

```diff
@@ -572,3 +578,43 @@ def validate_host(host, allowed_hosts):
     Return ``True`` for a valid host, ``False`` otherwise.
     """
     return any(pattern == '*' or is_same_domain(host, pattern) for pattern in allowed_hosts)
+
+
+class EnvironHeaders(CaseInsensitiveMapping):
+    # The following environ values will be ignored. For example, some WSGI
+    # servers will still send `HTTP_CONTENT_TYPE` even though is not part
+    # of PEP 333.
+    IGNORED_EXCEPTION_HEADERS = {'HTTP_CONTENT_TYPE', 'HTTP_CONTENT_LENGTH'}
+    # According to PEP 333, there are some special headers which are not
+    # prepended with HTTP_. These will remain untouched.
+    SPECIAL_UNCHANGED_HEADERS = {'CONTENT_TYPE', 'CONTENT_LENGTH'}
+    HTTP_PREFIX = 'HTTP_'
+
+    def __init__(self, environ):
+        header_name_generator = ((
+            self.parse_cgi_header_name(header_name), value
+        ) for header_name, value in environ.items())
+        headers = {
+            header: value for header, value in header_name_generator if header
+        }
+
+        super().__init__(headers)
+
+    @classmethod
+    def _style_header_name(cls, header_name):
+        return header_name.title()
```

Comment:
Does this need to be a separate method? Seems unnecessary to me.

Reply:
This can be extended by subclasses and overwritten, that's why I defined it as a different method.